### PR TITLE
Add exception handling for broken config files

### LINKED ISSUE #5667: Add a flag to disable config parsing
### Current problem

Here's a sample broken `.pylintrc`, placed in the root folder:

```ini
[pylint]
test = '%A'
```

This is what happens when I try to use pylint now:
```console
$ pylint myfile.py
Traceback (most recent call last):
  File "/Users/tusharsadhwani/code/marvin-python/venv3/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
  File "/Users/tusharsadhwani/code/marvin-python/venv3/lib/python3.10/site-packages/pylint/__init__.py", line 24, in run_pylint
    PylintRun(sys.argv[1:])
  File "/Users/tusharsadhwani/code/marvin-python/venv3/lib/python3.10/site-packages/pylint/lint/run.py", line 324, in __init__
    linter.load_config_file()
  File "/Users/tusharsadhwani/code/marvin-python/venv3/lib/python3.10/site-packages/pylint/config/option_manager_mixin.py", line 319, in load_config_file
    for option, value in parser.items(section):
  File "/usr/local/Cellar/python@3.10/3.10.1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 860, in items
    return [(option, value_getter(option)) for option in orig_keys]
  File "/usr/local/Cellar/python@3.10/3.10.1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 860, in <listcomp>
    return [(option, value_getter(option)) for option in orig_keys]
  File "/usr/local/Cellar/python@3.10/3.10.1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 856, in <lambda>
    value_getter = lambda option: self._interpolation.before_get(self,
  File "/usr/local/Cellar/python@3.10/3.10.1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 395, in before_get
    self._interpolate_some(parser, option, L, value, section, defaults, 1)
  File "/usr/local/Cellar/python@3.10/3.10.1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 442, in _interpolate_some
    raise InterpolationSyntaxError(
configparser.InterpolationSyntaxError: '%' must be followed by '%' or '(', found: "%A'"

$ pylint --help   
Traceback (most recent call last):
[...]
configparser.InterpolationSyntaxError: '%' must be followed by '%' or '(', found: "%A'"

$ pylint       
Traceback (most recent call last):
[...]
configparser.InterpolationSyntaxError: '%' must be followed by '%' or '(', found: "%A'"
```

Is it possible to add a flag that disables config parsing entirely, so that the pylint isn't completely broken when this happens? `--ignore-config-file` for example?

P.S. this is a feature I want, and I'd be willing to contribute it.

### PR DESCRIPTION
**Pull request by tusharsadhwani**
Extends the error handling of bad config files, to catch even more errors.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #5667

### CODE REVIEW

File: pylint/config/config_file_parser.py [hunk 1]
```diff
@@ -94,6 +94,10 @@ def parse_config_file(
...
-        if file_path.suffix == ".toml":
-            return self._parse_toml_file(file_path)
-        return self._parse_ini_file(file_path)
+        try:
+            if file_path.suffix == ".toml":
+                return self._parse_toml_file(file_path)
+            return self._parse_ini_file(file_path)
+        except (configparser.Error, tomllib.TOMLDecodeError) as e:
+            self.linter.add_message("config-parse-error", line=0, args=str(e))
+            return {}, []
```
Comment (Pierre-Sassoulas) [inline pylint/config/config_file_parser.py:103]: I think this should be handled inside ``_parse_toml_file`` and ``_parse_ini_file``?

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you @tusharsadhwani, the tests are on point, I have a nitpick concerning the place where we should catch the errors.

File: pylint/config/config_file_parser.py (hunk 1 above)
Comment (tusharsadhwani) [reply pylint/config/config_file_parser.py:103]: I think the intent is clearer this way: for any config failure (other than a missing pylint section in toml), raise this error.

There would also be a lot more basically duplicate try catch blocks if we make them more granular.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by DanielNoord** [comment]
Thanks @tusharsadhwani, very clean fix!

**Pull request merged**